common/flatpak-installation.c

```diff
@@ -991,6 +991,55 @@ _ostree_collection_ref_free0 (OstreeCollectionRef *ref)
   ostree_collection_ref_free (ref);
 }
 
+static gboolean
+flatpak_installation_check_installed_ref_missing_related_ref (FlatpakInstallation *self,
+                                                              const gchar         *remote_name,
+                                                              gchar               *full_ref,
+                                                              GCancellable        *cancellable)
+{
+  g_autoptr(FlatpakDir) dir  = NULL;
+  g_autoptr(GPtrArray) remote_related_refs = NULL;
+  g_autoptr(FlatpakRemoteState) state = NULL;
+  g_autoptr(GError) local_error = NULL;
+  guint j;
+
+  dir = flatpak_installation_get_dir (self, &local_error);
+  if (dir == NULL)
+    return FALSE;
+
+  state = flatpak_dir_get_remote_state_optional (dir, remote_name, FALSE, cancellable, &local_error);
```

Comment:
The loop in the function that calls check_installed_ref_missing_related_ref always is the same remote, so it could call this once and pass it every time, which will be much faster.

Reply:
Also can you move this entire function to flatpak-dir.c and make it not refer to FlatpakInstallation at all?

Reply:
Addressed.